data/dataset.py

```diff
@@ -142,35 +141,34 @@ def parse(cls, input):
 class Dataset:
     image_configs: dict[str, ImageConfig]
 
-    def __global_cfg(files):
+    def __global_cfg(fileset):
         cfgs = []
-        for file in files:
-            match os.path.basename(file):
-                case 'global.yaml' | 'global.yml':
-                    cfgs.append(ImageConfig.from_file(file))
+        
+        for cfgfile in ['global.yaml', 'global.yml']:
+            if cfgfile in fileset:
+                cfgs.append(ImageConfig.from_file(fileset[cfgfile]))
         return ImageConfig.fold(cfgs)
 
-    def __local_cfg(files):
+    def __local_cfg(fileset):
         cfgs = []
-        for file in files:
-            match os.path.basename(file):
-                case 'multiply.txt':
-                    cfgs.append(ImageConfig(multiply=read_float(file)))
-                case 'cond_dropout.txt':
-                    cfgs.append(ImageConfig(cond_dropout=read_float(file)))
-                case 'flip_p.txt':
-                    cfgs.append(ImageConfig(flip_p=read_float(file)))
-                case 'local.yaml' | 'local.yml':
-                    cfgs.append(ImageConfig.from_file(file))
+        if 'multiply.txt' in fileset:
+            cfgs.append(ImageConfig(multiply=read_float(fileset['multiply.txt'])))
+        if 'cond_dropout.txt' in fileset:
+            cfgs.append(ImageConfig(cond_dropout=read_float(fileset['cond_dropout.txt'])))
+        if 'flip_p.txt' in fileset:
+            cfgs.append(ImageConfig(flip_p=read_float(fileset['flip_p.txt'])))
+        if 'local.yaml' in fileset:
+            cfgs.append(ImageConfig.from_file(fileset['local.yaml']))
+        if 'local.yml' in fileset:
+            cfgs.append(ImageConfig.from_file(fileset['local.yml']))
         return ImageConfig.fold(cfgs)
 
-    def __sidecar_cfg(imagepath, files):
+    def __sidecar_cfg(imagepath, fileset):
         cfgs = []
-        for file in files:
```

Comment:
This is almost certainly the perfomance culprit, with an n^2 loop to resolve image sidecars.